### PR TITLE
feat: cors filter

### PR DESCRIPTION
**Pull request by yasinpalizban**
Supersedes #7253

**Description**
i add cors middleware to default filter CI4 and there is a config file which can easily modify.
all we have to do go to app/filter.php and uncooment  cors  that's all needs.

sysytem/filter/cors.php
app/config/cors.php
app/config/filter.php

others is not mine.

i ran this  commend vendor/bin/phpunit and get errors which is not my faults.

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [ ] Conforms to style guide

### CODE REVIEW

File: user_guide_src/source/libraries/uri.rst [hunk 1]
```diff
@@ -31,13 +31,14 @@ The Current URI
 ---------------
 
 Many times, all you really want is an object representing the current URL of this request.
-You can use one of the functions available in the **url_helper**:
+You can use one of the functions available in the :doc:`../helpers/url_helper`:
```
Comment (ddevsr) [inline user_guide_src/source/libraries/uri.rst:34]: This file

File: user_guide_src/source/libraries/uri/016.php [hunk 2]
```diff
@@ -2,5 +2,5 @@
 
 $uri = new \CodeIgniter\HTTP\URI('http://www.example.com/some/path');
 
-echo $uri->getPath();   // 'some/path'
-echo $uri->setPath('another/path')->getPath();  // 'another/path'
+echo $uri->getPath();                            // '/some/path'
+echo $uri->setPath('/another/path')->getPath();  // '/another/path'
```
Comment (ddevsr) [inline user_guide_src/source/libraries/uri/016.php:6]: And this file

**Review by ddevsr — CHANGES_REQUESTED** [review]
Remove changes in 2 file my mention

File: app/Database/Migrations/2018-01-24-102301_Some_migration.php [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        $this->db->table('foo')->insert([
+            'key' => 'foobar',
+        ]);
+    }
+
+    public function down()
+    {
+        $this->forge->dropTable('foo', true);
+    }
+}
```
Comment (ddevsr) [inline app/Database/Migrations/2018-01-24-102301_Some_migration.php:35]: This file not related to this PR

**Comment by kenjis** [comment]
As ddevsr says, please remove in this PR:
- app/Controllers/Home.php
- app/Database/Migrations/2018-01-24-102301_Some_migration.php
- user_guide_src/source/libraries/uri.rst
- user_guide_src/source/libraries/uri/016.php

**Comment by datamweb** [comment]
@yasinpalizban , this PR may need some changes. So it is better to commit [atomically](https://en.wikipedia.org/wiki/Atomic_commit#Atomic_commit_convention).

Also, there is **no need** to close PR, if you want to make changes, you can use the following command to apply changes in this PR after applying and committing it. 

Closing and opening new PR makes it difficult to follow up and review PR.

```console
git  push -f origin paliz
```

**Pull request closed without merging**